soc/arm/st_stm32/common/stm32_hsem.h

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2019 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DRIVERS_HSEM_STM32_HSEM_H_
+#define ZEPHYR_INCLUDE_DRIVERS_HSEM_STM32_HSEM_H_
+
+#include <soc.h>
+#include <kernel.h>
+
+#if defined(CONFIG_SOC_SERIES_STM32WBX) || defined(CONFIG_STM32H7_DUAL_CORE)
+/** HW semaphore Complement ID list defined in hw_conf.h from STM32WB
+ * and used also for H7 dualcore targets
+ */
+/** Index of the semaphore used to manage the entry Stop Mode procedure */
+#define CFG_HW_ENTRY_STOP_MODE_SEMID        4
+#define CFG_HW_ENTRY_STOP_MODE_MASK_SEMID   (1 << CFG_HW_ENTRY_STOP_MODE_SEMID)
+
+/** Index of the semaphore used to access the RCC and PWR */
+#define CFG_HW_RCC_SEMID                    3
+
+/** Index of the semaphore used to access the FLASH */
+#define CFG_HW_FLASH_SEMID                  2
+
+/** Index of the semaphore used to access the PKA */
+#define CFG_HW_PKA_SEMID                    1
+
+/** Index of the semaphore used to access the RNG */
+#define CFG_HW_RNG_SEMID                    0
+
+/** Index of the semaphore used to access GPIO */
+#define CFG_HW_GPIO_SEMID                   5
+
+/** Index of the semaphore used to access the EXTI */
+#define CFG_HW_EXTI_SEMID                   6
+
+#elif defined(CONFIG_SOC_SERIES_STM32MP1X)
+/** HW semaphore from STM32MP1
+ * EXTI and GPIO are inherited from STM32MP1 Linux
+ */
+/** Index of the semaphore used to manage the entry Stop Mode procedure */
+#define CFG_HW_ENTRY_STOP_MODE_SEMID        6
+#define CFG_HW_ENTRY_STOP_MODE_MASK_SEMID   (1 << CFG_HW_ENTRY_STOP_MODE_SEMID)
+
+/** Index of the semaphore used to access the RCC and PWR */
+#define CFG_HW_RCC_SEMID                    5
+
+/** Index of the semaphore used to access the FLASH */
+#define CFG_HW_FLASH_SEMID                  4
+
+/** Index of the semaphore used to access the PKA */
+#define CFG_HW_PKA_SEMID                    3
+
+/** Index of the semaphore used to access the RNG */
+#define CFG_HW_RNG_SEMID                    2
+
```

Comment:
please don't define semaphore 2 to 5 , this would means that the zephyr could access to these IPs, but this is not the case as reserved for the MPU usage. This could lead to corruption